### PR TITLE
Prepare stack for being integrated into nymea core

### PR DESCRIPTION
**Pull request by t-mon**

### CODE REVIEW

File: libnymea-zigbee/libnymea-zigbee.pro [hunk 1]
```diff
@@ -3,8 +3,14 @@ include(../config.pri)
 TARGET = nymea-zigbee1
 TEMPLATE = lib
 
-CONFIG += link_pkgconfig
-PKGCONFIG += libudev
+
+disable_udev {
```
Comment (mzanetti) [inline libnymea-zigbee/libnymea-zigbee.pro:7]: actually, thinking about this, even easier would be
```
CONFIG += link_pkgconfig
package_exists(libudev) {
    PKGCONFIG += libudev
    DEFINES += WITH_UDEV
}
```

Comment (t-mon) [reply libnymea-zigbee/libnymea-zigbee.pro:7]: my only concern: `package_exists` is not available for all qt verions, and if you don't need libudev, you don't need pkgconfig.

Comment (mzanetti) [reply libnymea-zigbee/libnymea-zigbee.pro:7]: packagesExist() I mean... What makes you believe it's not available?

Comment (t-mon) [reply libnymea-zigbee/libnymea-zigbee.pro:7]: ah, from pkg config, right...makes sense

Comment (t-mon) [reply libnymea-zigbee/libnymea-zigbee.pro]: fixed with last commit

Comment (mzanetti) [reply libnymea-zigbee/libnymea-zigbee.pro]: also aligns with cmake style, as the normal way to do it there is to 

`find_packages(name, VAR_IF_TRUE, VAR_IF_FALSE)` 
and then build the code around the outcome of that.

**Review by mzanetti — CHANGES_REQUESTED** [review]
There are some TODOs and commented code in the deconz backend (flush()). Please check again if can be done or dropped or needs to stay a TODO

ZigbeeBridgeController has a m_settingsDirectory. For one I think that should be private, not protected (perhaps with a protected getter() if really needed in the backend), but also please use QStandardPaths instead of /etc/ to initialize it or we'll have a problems in various platforms where /etc/ is not usable or we need to store settings somewhere else for factory reset reasons (snap, sue...)... I believe in the end this is set by the library user anyways but if you provide a default, it should be a sane default, and giving write-access to the backend seems not the best idea. Also the other protected members here should be revised if the backend really needs to have raw write access.

ZigbeeDataType::ZigbeeDataType(Zigbee::DataType dataType, const QByteArray &data): Is the todo there important?

ZigbeeDataType has a m_className which seems to have pretty-printed strings. Do we care for translations?

ZigbeeManufacturer: name related methods unused/incomplete. 

ZigbeeNetwork: I think this is a case where I wouldn't have used inheritance but instead composition (ZigbeeNetwork doing `new Backend()` instead of it "being" a Backend). Having the backend directly inherit ZigbeeNetwork (and probably the same goes for Interface and such) seems to give the backend too much influence over the whole thing. That results in very different behaviors of the entire stack between backends which makes debugging a lot harder (i.e. bugs happening only with one but not the other backend). I understand this feedback comes in quite late in the process. As you're thinning out the backend layers perhaps you want to consider this and perhaps can tie it down eventually.

ZigbeeDatabase: How future-proof do you think this is? Do you think we need a versioning mechanism? Can you think of upgrade-paths in case we need to change the db later without losing the users network information?

ZigbeeNetworkManager::availableNetworkTypes() returns hardcoded strings. Ideally it would be dynamic, depending on the backend types. Those are strings while down the layers it uses enums still (in this case we want strings I guess).

ZigbeeNetwork uses srand(). Not sure atm how much of a problem it really is in here, but srand is not threadsafe, so we should probably qsrand(). That one is deprecated for QRandomNumberGenerator which only exists of Qt 5.10 tho... In general both, srand() and qsrand() are not really a good thing to use as they use one global seed per process. If, for example, a plugin in nymea needs a secure random number and initializes it to something sophsiticated, and after that your zigbee code comes along along and re-initializes it with a timestamp... Admittely a somewhat theoretical attack vector, but there are enough examples of broken encryptions because of such mistake out there... Anyhow, I suggest qsrand() to at least avoid the possiblity to clash with other srand() calls in plugin threads, as well as forcing us to update to QRandromNumberGenerator with Qt6.

ZigbeeNode:
661: missing "a" in comment: "a bad stack"
712: Typo in comment: eiterh -> either

ZigbeeNodeEndpoint: 271: not sure if there's the need to if() a comment, lol. Kidding aside, seems like you forgot something there?

ZigbeeSecurityConfiguration: Where does that globalTrustCenterLinkKey come from? I probably just don't know what I'm talking about this, but please give it a minute of thought if we really want a hardcoded static key in the source code... It can be overridden, is this a default? Should it be randomized on first startup perhaps?

**Comment by mzanetti** [comment]
merged

**Pull request closed without merging**